core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -715,12 +734,40 @@ private[deploy] class Master(
     val numExecutors = coresPerExecutor.map { assignedCores / _ }.getOrElse(1)
     val coresToAssign = coresPerExecutor.getOrElse(assignedCores)
     for (i <- 1 to numExecutors) {
-      val exec = app.addExecutor(worker, coresToAssign)
+      val allocated = worker.acquireResources(app.desc.resourceReqsPerExecutor)
+      val exec = app.addExecutor(worker, coresToAssign, allocated)
       launchExecutor(worker, exec)
       app.state = ApplicationState.RUNNING
     }
   }
 
+  def hasEnoughResources(resourcesFree: Map[String, Int], resourceReqs: Map[String, Int])
```

Comment:
there is a function in TaskSchedulerImp.resourcesMeetTaskRequirements that basically does this as well, perhaps we can commonize.